hub/core/dataset/dataset.py

```diff
@@ -565,10 +577,18 @@ def commit(self, message: Optional[str] = None) -> str:
 
         Returns:
             str: the commit id of the stored commit that can be used to access the snapshot.
+
+        Raises:
+            Exception: if dataset is a filtered view.
         """
         return self._commit(message)
 
     def _commit(self, message: Optional[str] = None, hash: Optional[str] = None) -> str:
+        if getattr(self, "_is_filterd_view", False):
```

Comment:
why aren't we just adding self._is_filtered_view = False in init instead of getattr?